eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs

```diff
@@ -287,89 +298,80 @@ epochTransition = do
         { utxosDeposited =
             totalObligation
               adjustedCertState
-              (utxoSt' ^. utxosGovStateL)
+              (utxoSt' ^. utxosGovStateL) --  . proposalsGovStateL)
         }
     govSt = utxosGovState utxoSt''
     stakeDistr = credMap $ utxosStakeDistr utxoSt''
-    drepDistr = extractDRepDistr (es' ^. epochStateDRepDistrL)
-    ratEnv =
-      RatifyEnv
-        { reStakeDistr = stakeDistr
-        , reStakePoolDistr = stakePoolDistr
-        , reDRepDistr = drepDistr
-        , reCurrentEpoch = eNo - 1
-        , reDRepState = vstate ^. vsDRepsL
-        , reCommitteeState = vstate ^. vsCommitteeStateL
-        }
-    ratSig = RatifySignal . snapshotActions . prevGovSnapshots $ cgGovSnapshots govSt
-  RatifyState {rsRemoved, rsEnactState = rsEnactState'} <-
-    trans @(EraRule "RATIFY" era) $
-      TRC
-        ( ratEnv
-        , RatifyState
-            { rsRemoved = mempty
-            , rsEnactState = (govSt ^. cgEnactStateL) & ensTreasuryL .~ (es' ^. epochStateTreasuryL)
-            , rsDelayed = False
-            }
-        , ratSig
-        )
+    RatifyState {rsRemoved, rsEnactState} = extractDRepPulsingState (es1 ^. epochStateDRepPulsingStateL)
+  -- We are now finished with the pulser started at the last epoch boundary, so we need to
+  -- initialize a fresh DRep pulser, by computing the pulse size, and gathering the data we
+  -- will snapshot inside the pulser. We expect approximately 10*k-many blocks to be produced
+  -- each epoch. Therefore to safely and evenly space out the DRep calculation, we divide
+  -- the number of stake credentials by 8*k (8, rather than 10, to be sure we finish
+  -- two stability windows before the end of the epoch)
+  globals <- liftSTS ask
+  k <- liftSTS $ asks securityParameter -- Maximum number of blocks we are allowed to roll back
+  let currentProposals = govSt ^. cgProposalsL
+      newProposals = snapshotRemoveIds rsRemoved currentProposals
+  let stakeSize = Map.size stakeDistr
+      pulseSize = max 1 (ceiling (toInteger stakeSize % (8 * toInteger k)))
+      newPulser =
+        startDRepPulsingState
+          pulseSize
+          (dstate ^. dsUnifiedL)
+          stakeDistr
+          stakePoolDistr
+          (vstate ^. vsDRepsL)
+          (eNo + 1) -- Recall that the pulser will run the RATIFY rule in the next Epoch.
```

Comment:
I believe this is incorrect. 
```suggestion
          eNo
```

However, we can keep like this in this PR, I'll submit a separate PR fixing this with explanation why this should be the current epoch number